packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -442,67 +499,48 @@ export class ContextsManager {
     });
   }
 
+  dispatchTimer: NodeJS.Timeout | undefined;
   private dispatch(options: DispatchOptions): void {
-    if (options.contextsGeneralState) {
-      this.dispatchContextsGeneralState();
-    }
-    if (options.currentContextGeneralState) {
-      this.dispatchCurrentContextGeneralState();
-    }
-    if (options.resources?.pods) {
-      this.dispatchCurrentContextResource('pods');
-    }
-    if (options.resources?.deployments) {
-      this.dispatchCurrentContextResource('deployments');
-    }
+    clearTimeout(this.dispatchTimer);
+    this.dispatchTimer = setTimeout(() => {
+      if (options.contextsGeneralState) {
+        this.dispatchContextsGeneralState();
+      }
+      if (options.currentContextGeneralState) {
+        this.dispatchCurrentContextGeneralState();
+      }
+      if (options.resources?.pods) {
+        this.dispatchCurrentContextResource('pods');
+      }
+      if (options.resources?.deployments) {
+        this.dispatchCurrentContextResource('deployments');
+      }
+    }, dispatchTimeout);
```

Comment:
it looks like a lot of copy/paste for almost the same call
probably need to refactor to make less calls

Reply:
I refactored the last two ones. This will be useful when adding more informers

Reply:
thanks